types/src/crypto/asymmetric_key.rs

```diff
@@ -81,17 +81,20 @@ where
             return Err(Error::AsymmetricKey("too short".to_string()));
         }
 
-        let (tag_bytes, key_bytes) = input.as_ref().split_at(2);
+        let (tag_hex, key_hex) = input.as_ref().split_at(2);
 
-        let tag = checksummed_hex::decode(&tag_bytes)?;
-        let key_bytes = checksummed_hex::decode(&key_bytes)?;
+        let tag = checksummed_hex::decode(&tag_hex)?;
+        let key_bytes = checksummed_hex::decode(&key_hex)?;
 
         match tag[0] {
+            SYSTEM_TAG => Err(Error::AsymmetricKey(
+                "failed to decode from hex: cannot decode system variant".to_string(),
+            )),
```

Comment:
I think we should allow this. If someone tries to use the system key then it should fail any signature validation later. In Ethereum, a common practice for burning tokens is to send them to an address that no key exists for - 0x00000...